src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,49 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        robotMoveX(robot,toX);
+        robotMoveY(robot,toY);
+    }
+
+    public void robotMoveX(Robot robot, int toX) {
+        if (robot.getX() == toX) {
+            return;
+        }
+
+        if (robot.getX() < toX) {
+            while (!robot.getDirection().equals(Direction.RIGHT)) {
+                robot.turnRight();
+            }
+        } else {
+            while (!robot.getDirection().equals(Direction.LEFT)) {
+                robot.turnRight();
+            }
+        }
+
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+    }
+
+    public void robotMoveY(Robot robot, int toY) {
+        if (robot.getY() == toY) {
+            return;
+        }
+
+        if (robot.getY() < toY) {
+            while (!robot.getDirection().equals(Direction.UP)) {
+                robot.turnRight();
+            }
+        } else {
+            while (!robot.getDirection().equals(Direction.DOWN)) {
+                robot.turnRight();
+            }
+        }
+
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
     }
 }
 
```

Comment:
let's remove redundant empty lines at the end of the file

Reply:
Ok. Done